tofu/tests/tests01_geom/test_01_GG.py

```diff
@@ -1925,3 +1928,52 @@ def test24_is_visible(debug=2):
 
         print(are_vis)
         print(np.shape(are_vis))
+
+
+def test25_sa_integ_map(ves_poly=VPoly, debug=1):
+    import tofu.geom as tfg
+    import matplotlib.pyplot as plt
+
+    ves_norm = ves_poly[:, 1:] - ves_poly[:, :-1]
+    ves_norm = np.array([-ves_norm[1, :], ves_norm[0, :]])
+    ves_norm = ves_norm / np.sqrt(np.sum(ves_norm**2, axis=0))[np.newaxis, :]
+    print("ok at least this print ?")
+    if debug > 0:
+        # Visualisation:
+        ves = tfg.Ves(
+            Name="DebugVessel",
+            Poly=ves_poly,
+            Type="Tor",
+            Exp="Misc",
+            shot=0
+        )
+
+    part = np.array([[2.0, 2., 0.], [1.5, 0, 0], [2.5, 0, 0]], order='F').T
+    part_rad = np.r_[0.1, 0.1, 0.1]
+    rstep = zstep = phistep = 0.01
+    phistep = 0.5
+    RMinMax = np.array([np.min(ves_poly[0, :]), np.max(ves_poly[0, :])])
+    ZMinMax = np.array([np.min(ves_poly[1, :]), np.max(ves_poly[1, :])])
+
+    res = GG.compute_solid_angle_map(part, part_rad,
+                                     rstep, zstep, phistep,
+                                     RMinMax, ZMinMax,
+                                     ves_poly=ves_poly,
+                                     ves_norm=ves_norm,
+                                     )
+    pts, sa_map, ind, rdrdz = res
+
+    print(np.shape(sa_map))
+
+    if debug > 0:
+        fig = plt.figure(figsize=(14, 8))
+        ax = plt.subplot(121)
+        ax.plot(pts[0, :], pts[1, :], '.b')
+
+    # check size r,z, p
```

Comment:
Why not run these checks ?

Reply:
They were outdated, I updated it. Do I have your consent to change the python computation of the exact version to make a real check ?